### PR TITLE
docs: make nz-code-box to skip hydration

### PR DESCRIPTION
**Pull request by HyperLife1119**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #8510

image

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: scripts/site/_site/doc/app/share/codebox/codebox.component.html [hunk 1]
```diff
@@ -15,11 +15,9 @@
...
     <div class="code-box-title">
-      <a (click)="navigateToFragment()"
-      >{{ nzTitle }}
-        <a class="edit-button" [attr.href]="nzHref" target="_blank" rel="noopener noreferrer">
-          <span nz-icon nzType="edit"></span>
-        </a>
+      <a (click)="navigateToFragment()">{{ nzTitle }}</a>
+      <a class="edit-button" [attr.href]="nzHref" target="_blank" rel="noopener noreferrer">
+        <span nz-icon nzType="edit"></span>
       </a>
```
Comment (HyperLife1119) [inline scripts/site/_site/doc/app/share/codebox/codebox.component.html:21]: Browsers flatten nested anchor elements, which can cause the nodes to be rendered in a different actual position than their static position in the HTML file, causing the hydration to fail.

```html
<!-- file.html -->
<a>
  outer
  <a>inner</a>
</a>

<!-- after rendering -->
<a>outer</a>
<a>inner</a>
```

**Review by Nicoss54 — APPROVED** [review]
LGTM

**Comment by Jeeva-g2000** [comment]
Hello, will this pull request merge solve the issues with the official doc website?
Any update will be kindly appreciated... :)

**Comment by HyperLife1119** [comment]
> Hello, will this pull request merge solve the issues with the official doc website? Any update will be kindly appreciated... :)

You're right.

**Review by Laffery — APPROVED** [review]
LGTM

**Pull request merged**